src/character.cpp

```diff
@@ -1157,8 +1157,24 @@ bool Character::can_use( const item &it, const item &context ) const
     return true;
 }
 
-void Character::drop_inventory_overflow()
-{
+void Character::drop_invalid_inventory()
+{
+    bool dropped_liquid = false;
+    const_invslice invref = inv.const_slice();
```

Comment:
This does not need to be a separate varaible as it's only used once. You can iterate over the result of `const_slice` directly.

Btw. have you considered using `remove_items_with` and a filter that yields all liquid items instead of all this manual stuff? Something like this:
```C++
const auto items_removed = remove_items( []( const item &it ) { return it.get_current_phase() == LIQUID; } );
if( !items_removed.empty() ) {
    // todo: put the contents of items_removed on the map.
    add_msg("some liquids have been removed");
}
```

Reply:
Bleh I wanted this to work to make it super easy but the visitable goes digging through containers as well which I don't want to do because then it dumps out all the liquids in all your containers.